tests/test_vmec.py

```diff
@@ -913,3 +915,47 @@ def test_vmec_boundary_subspace(DummyStellarator):
     zbs_ref = np.atleast_2d(np.array([0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 1, 0, 0]))
     np.testing.assert_allclose(rbc_ref, np.abs(rbc) > tol)
     np.testing.assert_allclose(zbs_ref, np.abs(zbs) > tol)
+
+
+@pytest.mark.unit
+@pytest.mark.solve
+def test_write_vmec_input(TmpDir):
+    """Test generated VMEC input file gives the original equilibrium when solved."""
+    # write VMEC input file
+    fname = str(TmpDir.join("input.SOLOVEV"))
+    eq0 = get("SOLOVEV")
+    W0 = eq0.compute("W")["W"]
+    VMECIO.write_vmec_input(eq0, fname)
+
+    # read VMEC input file and override defaults to match desc/examples/SOLOVEV
+    ir = InputReader()
+    inputs = ir.parse_inputs(fname)
+    inputs[0]["spectral_indexing"] = "fringe"
+    inputs[0]["L"] = 24
+
+    # solve from VMEC input file
+    fam = EquilibriaFamily(inputs)
+    fam.solve_continuation(
+        objective=inputs[0]["objective"],
+        optimizer=inputs[0]["optimizer"],
+        pert_order=[inp["pert_order"] for inp in inputs],
+        ftol=[inp["ftol"] for inp in inputs],
+        xtol=[inp["xtol"] for inp in inputs],
+        gtol=[inp["gtol"] for inp in inputs],
+        maxiter=[inp["maxiter"] for inp in inputs],
+        verbose=2,
+    )
+    eq1 = fam[-1]
+    W1 = eq1.compute("W")["W"]
+
+    # check that solution matches original equilibrium
+    np.testing.assert_allclose(eq0.L, eq1.L)
+    np.testing.assert_allclose(eq0.M, eq1.M)
+    np.testing.assert_allclose(eq0.N, eq1.N)
+    np.testing.assert_allclose(eq0.NFP, eq1.NFP)
+    np.testing.assert_allclose(eq0.Psi, eq1.Psi)
+    np.testing.assert_allclose(eq0.p_l, eq1.p_l)
+    np.testing.assert_allclose(eq0.i_l, eq1.i_l)
+    np.testing.assert_allclose(eq0.Rb_lmn, eq1.Rb_lmn)
+    np.testing.assert_allclose(eq0.Zb_lmn, eq1.Zb_lmn)
+    np.testing.assert_allclose(W0, W1)
```

Comment:
this is the only part that is a "solution", the rest are just inputs. If the solution is being tested, testing R_lmn and Z_lmn and L_lmn or something would be appropriate

Reply:
I initially tried comparing the coefficients R_lmn, Z_lmn, and L_lmn but they did not match well. I think this is because of the freedom in $\lambda$, and it happens to converge to a different but equally valid solution. I think I checked the flux surfaces and they agreed -- should I add a plot comparison test to check that? Any ideas on other tests to add?

Reply:
maybe not a plot_comparison but a comparison of the area enclosed by each, like we do currently when we compare VMEC and DESC solutions, or a comparison of |B| strength at points on a grid (since the poloidal angles are the same right?)

And maybe I am a bit confused but what is different btwn these two eqs? are the surfaces not the same, and the profiles as well?

Reply:
I like the suggestion to compare the enclosed areas -- I'll try that, thanks. 

The two equilibria are supposed to be the same. I take the original equilibrium, generate a VMEC input file for it, and then solve in DESC from that VMEC input. So I am testing to see if that equilibrium matches the original one. But there is a freedom on lambda for the internal surfaces during the solve, so comparing the coefficients themselves isn't sufficient.

Reply:
Sure but I guess I am saying that this should be equivalent to running DESC twice with the same inputs? since despite the internal freedom we should get the same answer since the code is deterministic (barring numerical precision/hardware differences). 

You could just re-solve the SOLOVEV example then check it against the other one. It is really not necessary if you do area difference of B strength instead, and I get that if the SOLOVEV example is from an earlier version of the code or used a different objective than it may have slight differences, but just wanted to point it out that they should theoretically be the same, even the coefficients, if the solution method and inputs were the same

Reply:
The code is deterministic, but different continuation methods can lead to slightly different results. In our SOLOVEV example, we do a manual continuation method with 2 steps of different resolutions. When we convert from a VMEC input file in this test, we solve it in a single step for the final resolution. So that's why the coefficients don't match exactly. 

I added the area difference to the test and the solutions agree to a low tolerance.